lib/client.js

```diff
@@ -164,18 +165,29 @@ class Client extends EventEmitter {
         }
         this.socket.write(body)
       } else if (body && typeof body.pipe === 'function') {
+        const cleanup = eos(this.socket, err => {
```

Comment:
I'm a little unsure here what happens if the socket unexpectedly emits `'end'`. Does `Client` somewhere detect this and call `Client.destroy(err)`?

Reply:
Created issue https://github.com/mcollina/undici/issues/21